relay-server/src/services/project.rs

```diff
@@ -1126,21 +1126,51 @@ impl Project {
         outcome_aggregator: Addr<TrackOutcome>,
         buckets: Vec<Bucket>,
     ) -> Option<(Scoping, ProjectMetrics)> {
+        self.check_buckets_inner(outcome_aggregator, buckets)
+            .map_err(|e| {
+                match e {
+                    CheckBucketsError::ProjectExpired(len) => {
+                        relay_log::error!(
+                            tags.project_key = self.project_key.as_str(),
+                            "there is no project state: dropping {len} buckets"
+                        );
+                    }
+                    CheckBucketsError::ProjectDisabled(len) => {
+                        relay_log::debug!("dropping {len} buckets for disabled project");
+                    }
+
+                    CheckBucketsError::NoScoping(len) => {
+                        relay_log::error!(
+                            tags.project_key = self.project_key.as_str(),
+                            "there is no scoping: dropping {len} buckets"
+                        );
+                    }
+                    CheckBucketsError::RateLimited(len) => {
+                        relay_log::debug!("dropping {len} buckets due to rate limit");
+                    }
+                };
+                e
+            })
+            .ok()
+    }
+
+    fn check_buckets_inner(
```

Comment:
I created this helper method to make the result unit-testable (specific error instead of `None`).